### PR TITLE
few missing allocation failure checks and releases on error paths

### PR DESCRIPTION
**Pull request by ajmohan**
- Missing checks for allocation failure.
- releasing memory in few missing error paths

### CODE REVIEW

File: crypto/evp/bio_b64.c [hunk 1]
```diff
@@ -118,10 +118,15 @@ static int b64_new(BIO *bi)
...
     if (ctx == NULL)
-        return (0);
+        return 0;
 
     ctx->cont = 1;
     ctx->start = 1;
+    if (ctx->base64 == NULL) {
+        OPENSSL_free(ctx);
+        return 0;
+    }
```
Comment (mattcaswell) [inline crypto/evp/bio_b64.c]: Shouldn't this block be *after* the call to `EVP_ENCODE_CTX_new()`?

Comment (ajmohan) [inline crypto/evp/bio_b64.c]: oops. fixed it

**Comment by mattcaswell** [comment]
+1

File: crypto/bio/bio_err.c [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
  * Licensed under the OpenSSL license (the "License").  You may not use
  * this file except in compliance with the License.  You can obtain a copy
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
-*/
-
-/*
- * NOTE: this file was auto generated by the mkerr.pl script: any changes
- * made to it will be overwritten when the script next updates this file,
- * only reason strings will be preserved.
```
Comment (kroeckx) [inline crypto/bio/bio_err.c]: Is that automatically removed by the script?

Comment (kroeckx) [inline crypto/bio/bio_err.c]: So the script was recently modified and doesn't write that anymore ...

**Comment by kroeckx** [comment]
+1

File: crypto/bio/bio_err.c (hunk 2 above)
Comment (richsalz) [inline crypto/bio/bio_err.c]: this will hopefully get addressed by the WIP "cleanup" MR in our internal server.

**Comment by mattcaswell** [comment]
Merged. Thanks.

**Pull request closed without merging**